webapp/src/components/sidebar/sidebarSettingsMenu.tsx

```diff
@@ -26,6 +27,12 @@ const SidebarSettingsMenu = React.memo((props: Props) => {
         props.setWhiteLogo(whiteLogo)
     }
 
+    const [randomIcons, setRandomIcons] = useState(UserSettings.prefillRandomIcons)
+    const toggleRandomIcons = () => {
+        UserSettings.prefillRandomIcons = !UserSettings.prefillRandomIcons
+        setRandomIcons(!randomIcons)
+    }
```

Comment:
This is a little odd because the setting state is maintained in two separate spots (`UserSettings` and `randomIcons`) but I couldn't find a better way to cause the component to re-render when changing the setting. Would be happy to hear recommendations on this.

Reply:
Yes, probably we can move this into its own hook and make the concept of "userSettings" something that is properly encapsulated. But for now I think is ok as is now.